packages/tldraw/src/state/StateManager/StateManager.ts

```diff
@@ -138,6 +168,22 @@ export class StateManager<T extends Record<string, any>> {
     }
   }
 
+  /**
+   * Save the setting to localStorage
+   */
+  protected persistSetting = (patch: Patch<U>, id?: string) => {
+    if (this._status !== 'ready') return
+    if (this.onPersist) {
+      this.onPersist(this._state, patch, id)
```

Comment:
On second thought, since we're passing `this` to `callbacks.onPersist`, the change would be detectable. 

Still feels like it's kind of overriding the documented purpose of `onPersist` though, since that's for persisting state changes but settings are no longer part of the state. And having `TldrawPatch` be two entirely different object types adds might add more complexity than it's worth.

Just some thoughts though, I don't really know how this system works yet 😆

Reply:
you're right we don't need it!